src/zenml/integrations/kserve/services/kserve_deployment.py

```diff
@@ -0,0 +1,568 @@
+                container = init_containers[0]
+
+            logger.info(
+                f"Retrieving logs for pod: `{pod_name}` and container "
+                f"`{container}` in namespace `{namespace}`"
+            )
+            response = client.core_api.read_namespaced_pod_log(
+                name=pod_name,
+                namespace=namespace,
+                container=container,
+                follow=follow,
+                tail_lines=tail,
+                _preload_content=False,
+            )
+        except k8s_client.rest.ApiException as e:
+            logger.error(
+                "Exception when fetching logs for InferenceService resource "
+                "%s: %s",
+                name,
+                str(e),
+            )
+            raise Exception(
+                f"Unexpected exception when fetching logs for InferenceService "
+                f"resource: {name}"
+            ) from e
+
+        try:
+            while True:
+                line = response.readline().decode("utf-8").rstrip("\n")
+                if not line:
+                    return
+                stop = yield line
+                if stop:
+                    return
+        finally:
+            response.release_conn()
+
+    def get_logs(
+        self, follow: bool = False, tail: Optional[int] = None
+    ) -> Generator[str, bool, None]:
+        """Retrieve the logs from the remote KServe inference service instance.
+
+        Args:
+            follow: if True, the logs will be streamed as they are written.
+            tail: only retrieve the last NUM lines of log output.
+
+        Returns:
+            A generator that can be accessed to get the service logs.
+        """
+        return self._get_deployment_logs(
+            self.crd_name,
+            follow=follow,
+            tail=tail,
+        )
+
+    @property
+    def prediction_url(self) -> Optional[str]:
+        """The prediction URI exposed by the prediction service.
+
+        Returns:
+            The prediction URI exposed by the prediction service, or None if
+            the service is not yet ready.
+        """
+        if not self.is_running:
+            return None
+
+        model_deployer = self._get_model_deployer()
+        model_name = self._get_kubernetes_labels().get("model_name") or "mnist"
+        return os.path.join(
+            model_deployer.base_url,
+            "v1/models",
+            f"{model_name}:predict",
+        )
+
+    @property
+    def prediction_hostname(self) -> Optional[str]:
+        """The prediction hostname exposed by the prediction service.
+
+        Returns:
+            The prediction hostname exposed by the prediction service status
+            that will be used in the headers of the prediction request.
+        """
+        if not self.is_running:
+            return None
+
+        namespace = self._get_namespace()
+        return f"{self.crd_name}.{namespace}.example.com"
```

Comment:
Ah, that should be one of the parameters in the model deployer! it's by default example.com in the normal KServe installation/configuration and changing that to custom stuff requires some additional processes from the user! it's now optional so if the user has their own custom stuff that will be used otherwise it's example.com